### PR TITLE
Fix apikey propagation

### PR DESCRIPTION
**Pull request by jazpenayo**

### CODE REVIEW

File: src/enviarApi.js [hunk 1]
```diff
@@ -18,9 +18,6 @@ export default class EnviarApi extends Component {
 
 
   enviar() {
-    this.setState(state => ({
-      estado: !state.estado
-    }));
     console.log("ENVIAR");
     console.log(this);
   }
```
Comment (tchx84) [inline src/enviarApi.js]: This function is not really doing anything, let's get rid of it?

File: src/source.js [hunk 2]
```diff
@@ -8,19 +8,24 @@ export default class Source extends Component {
...
-    this.state = { estado: true };
-
+    this.state = { apikey: "" };
+    this.handleApiKeyChange = this.handleApiKeyChange.bind(this);
 
   }
 
-
+  handleApiKeyChange(apikey) {
+    console.log("handleApiKeyChange ",apikey);
```
Comment (tchx84) [inline src/source.js]: let's get of this as well..

File: src/source.js [hunk 3]
```diff
@@ -8,19 +8,24 @@ export default class Source extends Component {
...
+    this.setState({apikey});
+  } 
 
   render() {
+    const apikey = this.state.apikey; 
+    console.log("render ",apikey);
```
Comment (tchx84) [inline src/source.js]: Let's get rid of this as well..

File: src/sourceList.js [hunk 4]
```diff
@@ -10,26 +9,26 @@ export default class SourceList extends Component {
         this.state = {
             sources: []
         }
+
```
Comment (tchx84) [inline src/sourceList.js]: No need for this empty line

File: src/sourceList.js [hunk 5]
```diff
@@ -10,26 +9,26 @@ export default class SourceList extends Component {
...
     }
 
-    state = {
-        sources: []
-    };
-    apiKey = {this.props.apiKey}
 
     componentDidMount() {
+        console.log("componentDidMount ",this.props.apikey);
```
Comment (tchx84) [inline src/sourceList.js]: this too

**Review by tchx84 — CHANGES_REQUESTED** [review]

**Comment by jazpenayo** [comment]
Ready

**Review by tchx84 — APPROVED** [review]

**Pull request merged**